File: mocha-game/src/main/java/mocha/game/world/entity/movement/command/EntityMoveCommand.java
package mocha.game.world.entity.movement.command;

import lombok.AllArgsConstructor;
import lombok.Data;
import mocha.game.command.Command;
import mocha.game.world.Direction;
import mocha.game.world.Location;

@Data
@AllArgsConstructor
public class EntityMoveCommand implements Command {

  private int entityId;
  private Location location;
  private Direction direction;
  private int xOffset;
  private int yOffset;

}

File: mocha-game/src/main/java/mocha/game/world/entity/movement/command/EntityMoveCommand.kt
package mocha.game.world.entity.movement.command

import mocha.game.command.Command
import mocha.game.world.Direction
import mocha.game.world.Location

data class EntityMoveCommand(
        var entityId: Int,
        val location: Location,
        val direction: Direction,
        val xOffset: Int,
        val yOffset: Int
) : Command
